runner/src/main.rs

```diff
@@ -285,15 +275,15 @@ fn build_server(opt: &BuildServer) -> Step {
                         "-Zunstable-options".to_string(),
                         ...match opt.server_variant {
                             ServerVariant::Logless => vec!["--no-default-features".to_string(),
-                                format!("--target={}", opt.server_rust_target.as_deref().unwrap_or(DEFAULT_SERVER_RUST_TARGET)),
+                                format!("--target={}", opt.server_rust_target.as_deref().unwrap_or(&std::env::var("DEFAULT_SERVER_RUST_TARGET").unwrap())),
```

Comment:
I am not convinced this approach is worth the effort of passing these as env variables that are set in the build script. What is the problem that it solves, vs for instance having a method in this file that returns the appropriate value for each of these strings? Is it because `const::ARCH` is only available at build time? Either way, it would be great to have it written down, for future developers touching this code.

Reply:
Yep, comments comments and comments needed: The long and short of it is that the structopt default_values are insufficent when the defaults change with the platform on which the system is built. By taking the specifics of the cludge out of the main code and into the build script, it keeps the rest of it generic. That means any new platform can tweak in the build.rs without having to fiddle with the main runner codebase. That particularly egrigious line is due to the type-coercion needed here for String to &str, and the intent for a deliberate failure if the DEFAULT_SERVER_TARGET is unset by the build script, which is the case only on an unrecognised platform (it will blow up on the unwrap).

Reply:
If I wrote a specific structopt parser for the strings here, I think it could improve it further, and would remove the nasty hackage here.